hack/bazel-fmt.sh

```diff
@@ -23,7 +23,6 @@ source hack/common.sh
 source hack/config.sh
 
 bazel run \
-    --config=${ARCHITECTURE} \
```

Comment:
> I think we have to remove `--config` here for all commands in this file. They all need to match the node architecture. In general I wonder if we should keep specifying a config, so that for the main architectures we are not thrashing the analysis cache for non-crosscompile-builds. for cross-compile builds I am less concerned if the build time is a little bit higher.
> 
> Would you mind checking the difference in speed and checking if for non-crosscompile builds you see bazel logging that it is dumping the analysis cache?

Ok, I get your idea, but I do not see any output related to cache in bazel logging, like following:
```
# make bazel-build
hack/dockerized "export BUILD_ARCH= && hack/bazel-fmt.sh && hack/bazel-build.sh"
selecting docker as container runtime
go version go1.13.14 linux/amd64

go version go1.13.14 linux/amd64
INFO: Build options --action_env and --platforms have changed, discarding analysis cache.
INFO: Analyzed target //vendor/mvdan.cc/sh/v3/cmd/shfmt:shfmt (74 packages loaded, 6979 targets configured).
INFO: Found 1 target...
Target //vendor/mvdan.cc/sh/v3/cmd/shfmt:shfmt up-to-date:
  bazel-bin/vendor/mvdan.cc/sh/v3/cmd/shfmt/shfmt_/shfmt
INFO: Elapsed time: 1.692s, Critical Path: 0.79s
INFO: 1 process: 1 internal.
INFO: Build completed successfully, 1 total action
INFO: Build completed successfully, 1 total action
INFO: Build option --platforms has changed, discarding analysis cache.
INFO: Analyzed target //:gazelle (29 packages loaded, 6757 targets configured).
INFO: Found 1 target...
Target //:gazelle up-to-date:
  bazel-bin/gazelle-runner.bash
  bazel-bin/gazelle
INFO: Elapsed time: 0.345s, Critical Path: 0.06s
INFO: 1 process: 1 internal.
INFO: Build completed successfully, 1 total action
INFO: Build completed successfully, 1 total action
INFO: Analyzed target //:goimports (14 packages loaded, 255 targets configured).
INFO: Found 1 target...
Target //:goimports up-to-date:
  bazel-bin/goimports.bash
INFO: Elapsed time: 0.295s, Critical Path: 0.06s
INFO: 1 process: 1 internal.
INFO: Build completed successfully, 1 total action
INFO: Build completed successfully, 1 total action
INFO: Analyzed target //:buildifier (67 packages loaded, 784 targets configured).
INFO: Found 1 target...
Target //:buildifier up-to-date:
  bazel-bin/buildifier.bash
INFO: Elapsed time: 0.278s, Critical Path: 0.06s
INFO: 1 process: 1 internal.
INFO: Build completed successfully, 1 total action
INFO: Build completed successfully, 1 total action
INFO: Analyzed 61 targets (1193 packages loaded, 10095 targets configured).
INFO: Found 61 targets...
INFO: From ImageLayer cmd/virt-launcher/version-container-layer.tar:
Duplicate file in archive: ./etc/libvirt/libvirtd.conf, picking first occurrence
Duplicate file in archive: ./etc/libvirt/qemu.conf, picking first occurrence
Duplicate file in archive: ./usr/share/man/man3/nfs4_uid_to_name.3.gz, picking first occurrence
Duplicate file in archive: ./etc/group, picking first occurrence
Duplicate file in archive: ./etc/passwd, picking first occurrence
Duplicate file in archive: ./usr/share/licenses/systemd/LICENSE.LGPL2.1, picking first occurrence
INFO: From ImageLayer cmd/virt-handler/version-container-layer.tar:
Duplicate file in archive: ./usr/share/man/man3/nfs4_uid_to_name.3.gz, picking first occurrence
Duplicate file in archive: ./etc/group, picking first occurrence
Duplicate file in archive: ./etc/passwd, picking first occurrence
Duplicate file in archive: ./usr/share/licenses/systemd/LICENSE.LGPL2.1, picking first occurrence
INFO: Elapsed time: 6.861s, Critical Path: 5.19s
INFO: 122 processes: 32 internal, 90 processwrapper-sandbox.
INFO: Build completed successfully, 122 total actions
INFO: Analyzed target //:build-dump (0 packages loaded, 1 target configured).
INFO: Found 1 target...
Target //:build-dump up-to-date:
  bazel-bin/dump-copier
INFO: Elapsed time: 0.225s, Critical Path: 0.05s
INFO: 2 processes: 1 internal, 1 processwrapper-sandbox.
INFO: Build completed successfully, 2 total actions
INFO: Build completed successfully, 2 total actions
INFO: Analyzed target //:build-virtctl (0 packages loaded, 1 target configured).
INFO: Found 1 target...
Target //:build-virtctl up-to-date:
  bazel-bin/virtctl-copier
INFO: Elapsed time: 0.294s, Critical Path: 0.06s
INFO: 2 processes: 1 internal, 1 processwrapper-sandbox.
INFO: Build completed successfully, 2 total actions
INFO: Build completed successfully, 2 total actions
INFO: Analyzed target //:build-virtctl-amd64 (0 packages loaded, 1 target configured).
INFO: Found 1 target...
Target //:build-virtctl-amd64 up-to-date:
  bazel-bin/virtctl-copier-amd64
INFO: Elapsed time: 0.449s, Critical Path: 0.21s
INFO: 2 processes: 1 internal, 1 processwrapper-sandbox.
INFO: Build completed successfully, 2 total actions
INFO: Build completed successfully, 2 total actions
INFO: Analyzed target //:build-virtctl-darwin (0 packages loaded, 1 target configured).
INFO: Found 1 target...
Target //:build-virtctl-darwin up-to-date:
  bazel-bin/virtctl-copier-darwin
INFO: Elapsed time: 0.567s, Critical Path: 0.37s
INFO: 2 processes: 1 internal, 1 processwrapper-sandbox.
INFO: Build completed successfully, 2 total actions
INFO: Build completed successfully, 2 total actions
INFO: Analyzed target //:build-virtctl-windows (0 packages loaded, 1 target configured).
INFO: Found 1 target...
Target //:build-virtctl-windows up-to-date:
  bazel-bin/virtctl-copier-windows
INFO: Elapsed time: 0.742s, Critical Path: 0.34s
INFO: 2 processes: 1 internal, 1 processwrapper-sandbox.
INFO: Build completed successfully, 2 total actions
INFO: Build completed successfully, 2 total actions
```

Reply:
That is the line I mean:

```
INFO: Build option --platforms has changed, discarding analysis cache.
```

Reply:
Here is my test scenario:
1. run non-crossbuild for two times, and record the time cost for the second one.
2. run crossbuild then run non-crossbuild, and record the time cost for the second one.

0m17.329s vs 3m28.490s

It is true, the cache has a great influence on compile speed.

Reply:
It is fine for now that the cross-compile takes a longer time. I am more intersted in the `0m17.329s`. If you still have the `--config` set on the tooling binaries it may again much faster than `0m17.329s`. That would be the interesting number.